pkg/controller/nodes/executor.go

```diff
@@ -792,8 +797,20 @@ func (c *nodeExecutor) AbortHandler(ctx context.Context, execContext executors.E
 		if err != nil {
 			return err
 		}
+		nodeExecutionId := &core.NodeExecutionIdentifier{
+			ExecutionId: nCtx.NodeExecutionMetadata().GetNodeExecutionID().ExecutionId,
+			NodeId: nCtx.NodeExecutionMetadata().GetNodeExecutionID().NodeId,
+		}
+		if nCtx.ExecutionContext().GetEventVersion() != v1alpha1.V0 {
+			currentNodeUniqueID, err := common.GenerateUniqueId(nCtx.ExecutionContext().GetParentInfo(), nodeExecutionId.NodeId)
+			if err != nil {
+				return err
+			}
+			nodeExecutionId.NodeId = currentNodeUniqueID
+		}
+
 		err = c.IdempotentRecordEvent(ctx, &event.NodeExecutionEvent{
```

Comment:
do we not need to pass the node name as well?

Reply:
Nope. this is abort. So only status is updated.

Reply:
but still should we not be consistent?

Reply:
@kumare3 We do not populate immutable fields here. I am trying to be consistent here and think it's right way.